src/pack.c

```diff
@@ -934,18 +934,20 @@ git_off_t get_delta_base(
 	if (type == GIT_OBJ_OFS_DELTA) {
 		unsigned used = 0;
 		unsigned char c = base_info[used++];
-		base_offset = c & 127;
+		size_t unsigned_base_offset = c & 127;
 		while (c & 128) {
 			if (left <= used)
 				return GIT_EBUFS;
-			base_offset += 1;
-			if (!base_offset || MSB(base_offset, 7))
+			unsigned_base_offset += 1;
+			if (!unsigned_base_offset || MSB(unsigned_base_offset, 7))
 				return 0; /* overflow */
 			c = base_info[used++];
-			base_offset = (base_offset << 7) + (c & 127);
+			unsigned_base_offset = (unsigned_base_offset << 7) + (c & 127);
 		}
-		base_offset = delta_obj_offset - base_offset;
-		if (base_offset <= 0 || base_offset >= delta_obj_offset)
+		if ((size_t)delta_obj_offset <= unsigned_base_offset)
+			return 0; /* out of bound */
+		base_offset = delta_obj_offset - unsigned_base_offset;
+		if (base_offset >= delta_obj_offset)
```

Comment:
This cannot ever be true:

base_offset >= delta_obj_offset
<=> delta_obj_offset - unsigned_base_offset  >= delta_obj_offset

The above could only be true iff unsinged_base_offset is greater than delta_obj_offset or 0, where the first case was filtered out in line 947 and the second case shouldn't ever happen. So you could just reformulate this condition as unsingned_base_offset == 0.

Reply:
Done